src/components/UserProfile/index.jsx

```diff
@@ -44,11 +44,25 @@ class UserProfile extends React.Component {
     this.props.openEditableField(fieldName);
   }
 
-  onSave(fieldName, value) {
+  onSave(fieldName, value, visibility) {
+    const {
+      value: fieldValue,
+      visibility: fieldVisibility,
+    } = this.state[fieldName];
+
     const userAccountData = {
-      [fieldName]: value || this.state[fieldName].value,
+      [fieldName]: value || fieldValue,
```

Comment:
We spoke in person, but for posterity: components that need to manage their own dirty state for whatever reason (in our current case social links) can provide the value directly to onSave rather than saving what's in the parent component state.